api.go

```diff
@@ -0,0 +1,67 @@
+package customerio
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type APIClient struct {
+	Key    string
+	URL    string
+	Client *http.Client
+}
+
+// NewAPIClient prepares a client for use with the Customer.io API, see: https://customer.io/docs/api/#apicoreintroduction
+// using an App API Key from https://fly.customer.io/settings/api_credentials?keyType=app
+func NewAPIClient(key string) *APIClient {
+	return &APIClient{
+		Key:    key,
+		Client: http.DefaultClient,
+		URL:    "https://api.customer.io",
+	}
+}
+
+func (c *APIClient) doRequest(ctx context.Context, verb, requestPath string, body interface{}) (*http.Response, []byte, error) {
+
+	b, err := json.Marshal(body)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := http.NewRequest("POST", c.URL+requestPath, bytes.NewBuffer(b))
```

Comment:
This should use https://golang.org/pkg/net/http/#NewRequestWithContext

Reply:
are we planning on fixing this before shipping?